base/missing.jl

```diff
@@ -107,16 +107,19 @@ max(::Missing, ::Any)     = missing
 max(::Any,     ::Missing) = missing
 
 # Rounding and related functions
-for f in (:(ceil), :(floor), :(round), :(trunc))
+round(::Missing, ::RoundingMode=RoundNearest; sigdigits::Integer=0, digits::Integer=0, base::Integer=0) = missing
+round(::Type{>:Missing}, ::Missing, ::RoundingMode=RoundNearest) = missing
+round(::Type{T}, ::Missing, ::RoundingMode=RoundNearest) where {T} =
+    throw(MissingException("cannot convert a missing value to type $T: use Union{$T, Missing} instead"))
+round(::Type{T}, x::Any, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+# to fix ambiguities
+round(::Type{T}, x::Rational, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+round(::Type{T}, x::Rational{Bool}, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+
+for f in (:(ceil), :(floor), :(trunc))
     @eval begin
-        ($f)(::Missing, digits::Integer=0, base::Integer=0) = missing
-        ($f)(::Type{>:Missing}, ::Missing) = missing
-        ($f)(::Type{T}, ::Missing) where {T} =
-            throw(MissingException("cannot convert a missing value to type $T: use Union{$T, Missing} instead"))
-        ($f)(::Type{T}, x::Any) where {T>:Missing} = $f(nonmissingtype(T), x)
-        # to fix ambiguities
-        ($f)(::Type{T}, x::Rational) where {T>:Missing} = $f(nonmissingtype(T), x)
-        ($f)(::Type{T}, x::Rational{Bool}) where {T>:Missing} = $f(nonmissingtype(T), x)
+        $(f)(::Missing; kwargs...) = round(T, missing; kwargs...)
```

Comment:
That's still completely incorrect. Please run tests locally before pushing, as the full test suite is run on 6 workers for each push, taking hours of CPU time.

Reply:
Sorry! I didn't expect this to turn into so much work and have been doing this in a combination of REPL and editing on Github! Have run the code locally again - seems to work for me.

Reply:
I canceled them manually. GitHub editing should really be reserved to trivial changes. Every time I've used it for more complex things I've regretted it.

Reply:
Yeah... this started off simple... guess I should have switched methods when it got complicated!